service/controller/v13/encrypter/vault/vault.go

```diff
@@ -174,20 +190,42 @@ func (e *Encrypter) CurrentState(ctx context.Context, customObject v1alpha1.AWSC
 func (e *Encrypter) DesiredState(ctx context.Context, customObject v1alpha1.AWSConfig) (encrypter.EncryptionKeyState, error) {
 	state := encrypter.EncryptionKeyState{}
 
-	keyName, err := e.EncryptionKey(ctx, customObject)
-	if err != nil {
-		return state, microerror.Mask(err)
-	}
+	keyName := e.keyName(customObject)
 
 	state.KeyName = keyName
 
 	return state, nil
 }
 
 func (e *Encrypter) EncryptionKey(ctx context.Context, customObject v1alpha1.AWSConfig) (string, error) {
-	clusterID := key.ClusterID(customObject)
+	err := e.ensureToken()
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	keyName := e.keyName(customObject)
+
+	p := path.Join("transit", "keys", keyName)
+
+	req, err := e.newRequest("GET", p)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	resp, err := e.httpClient.Do(req)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+	defer resp.Body.Close()
 
-	return clusterID, nil
+	if resp.StatusCode == http.StatusNotFound {
+		return "", microerror.Mask(keyNotFoundError)
+	} else if resp.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(resp.Body)
+		return "", microerror.Maskf(invalidHTTPStatusCodeError, "want 200, got %d, response body: %q", resp.StatusCode, body)
```

Comment:
Again nitpicking but you could also use `http.StatusOK` in the error message.

Reply:
Good point, done.